### PR TITLE
docs: methods update

### PR DESCRIPTION
**Pull request by FireboltCorey**
Documentation updates as follows, per Stepan:

Staged here: https://firebolt-sdk.readthedocs-hosted.com/en/latest/index.html

**Client**

-   remove resource_manager_hooks
-   auth should be extended 
    -   auth.username_password.UsernamePassword class
    -   auth.token.Token class

**Common**

-   Remove everything except settings
    -   In settings leave only first 6 attributes in description
- Expose Utils.exceptions

**Db**

-   Add `connect` function and put it to the top This does not have a class under it. Should it?
    -   Add a note to Connection that it shouldn't be created directly but rather with the `connect` function 

**Async_db**

-   Put connect function to the top
    -   Add a note to Connection that it shouldn't be created directly but rather with the `connect` function 

-   remove Async_db.util
-   remove firebolt.async_db.cursor.check_not_closed
-   remove firebolt.async_db.cursor.check_query_executed
-   remove firebolt.async_db.connection.async_connect_factory
-   remove firebolt.async_db.connection.OverriddenHttpBackend

**Model**

-   remove firebolt.model.database.FieldMask
-   remove firebolt.model.engine.FieldMask
-   remove firebolt.model.engine.check_attached_to_database
-   remove firebolt.model.engine.wait

**Service**

-   Looks fine

### CODE REVIEW

File: docsrc/firebolt.client.rst [hunk 1]
```diff
@@ -2,16 +2,22 @@
 Client
```
Comment (stepansergeevitch) [inline docsrc/firebolt.client.rst:2]: Autogenerated piece for `UsernamePassword` and `Token` doesn't look properly as well, similarly to `firebolt.async_db.connect`

File: docsrc/firebolt.db.rst [hunk 2]
```diff
@@ -2,13 +2,26 @@
 Db
```
Comment (stepansergeevitch) [inline docsrc/firebolt.db.rst:2]: Mostly same comments as for async_db

File: docsrc/firebolt.model.rst [hunk 3]
```diff
@@ -9,7 +9,7 @@ Model.database
 
 .. automodule:: firebolt.model.database
```
Comment (stepansergeevitch) [inline docsrc/firebolt.model.rst:10]: Can we rename every module header in such way:
`Model.engine_revision -> Engine revision`

File: docsrc/firebolt.async_db.rst [hunk 4]
```diff
@@ -2,14 +2,27 @@
...
+Async\_db.connect
+------------------------------------
+
+.. automodule:: firebolt.async_db.connect
+   :members:
+   :inherited-members:
+   :undoc-members:
+   :show-inheritance:
 
 Async\_db.connection
```
[the remark this answers is not in the document]
Comment (FireboltCorey) [reply docsrc/firebolt.async_db.rst:16]: I think you have to mark attributes as private in the code. Something like the example below.

class MyClass(object):
    """A class"""
    attr1 = 42  #: Attribute 1.
    attr2 = 43  #: Attribute 2.
    _private1 = 44  #: Private attribute 1.
    _private2 = 45  #: Private attribute 2.

File: docsrc/firebolt.async_db.rst [hunk 5]
```diff
@@ -2,14 +2,27 @@
 Async\_db
 ==========================
 
-The async_db package enables connecting to a Firebolt database for asynchronous queries. 
+The async_db package enables connecting to a Firebolt database for asynchronous queries.
+
+Async\_db.connect
+------------------------------------
+
+.. automodule:: firebolt.async_db.connect
```
[the remark this answers is not in the document]
Comment (FireboltCorey) [reply docsrc/firebolt.async_db.rst:10]: I changed the inheritance, other than that it's formatted like the others. See if that helps and if not, I'll have to investigate further.

File: docsrc/firebolt.async_db.rst [hunk 6]
```diff
@@ -2,14 +2,27 @@
 Async\_db
```
[the remark this answers is not in the document]
Comment (FireboltCorey) [reply docsrc/firebolt.async_db.rst:2]: Changed here and in firebolt.db.rst

**Review by stepansergeevitch — APPROVED** [review]
I still have some minor comments, mostly regarding the appearance. But I think those can be fixed in a separate PR. For now LGTM

**Pull request merged**